src/streamlink/plugins/rumble.py

```diff
@@ -0,0 +1,148 @@
+"""
+$description Rumble is an online video platform founded in October 2013 by Chris Pavlovski as an alternative to YouTube.
+$url rumble.com
+$type live
+$metadata id
+$metadata author
+$metadata title
+"""
+
+import logging
+import re
+from typing import Dict
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream, HTTPStream
+
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(
+    re.compile(
+        r"https?://rumble\.com/c/(?P<channel>[^/?]+)",
+    ),
+    name="channel",
+)
+@pluginmatcher(
+    re.compile(
+        r"https?://rumble\.com/(?P<room_id>[^/.]+)\.html",
+    ),
+    name="live",
+)
+class Rumble(Plugin):
+    QUALITY_WEIGHTS: Dict[str, int] = {}
+
+    _URL_WEB_LIVE = "https://rumble.com/c/{channel}/livestreams"
+    _URL_API_LIVE_DETAIL = "https://rumble.com/embedJS/u3/"
+
+    _PARAM_VER = 2
+    _PARAM_REQUEST = "video"
+
+    _SCHEMA_STREAM = {
+        str: validate.all(
+            {
+                "url": validate.url(),
+                "meta": {"bitrate": int},
+            },
+            validate.union_get("url", ("meta", "bitrate")),
+        ),
+    }
+
+    @classmethod
+    def stream_weight(cls, key):
+        weight = cls.QUALITY_WEIGHTS.get(key)
+        if weight:
+            return weight, key
+
+        return super().stream_weight(key)
+
+    def _get_streams(self):
+        if self.matches["channel"]:
+            channel = self.match["channel"]
+            live_uri = self.session.http.get(
+                self._URL_WEB_LIVE.format(channel=channel),
+                schema=validate.Schema(
+                    validate.parse_html(),
+                    validate.xml_xpath_string(
+                        "/html/body/main/section/ol/div[1]/div[2][@class='videostream__footer "
+                        + "videostream__footer--live']/a/@href",
+                    ),
+                ),
+                timeout=5,
+            )
+            if not live_uri:
+                return
+            room_url = f"https://rumble.com{live_uri}"
```

Comment:
That's an unsafe URL concatenation which must be avoided when random data is involved.
See `streamlink.utils.url.url_concat` or `urllib.parse.{urlunparse,urljoin}`.